airflow/providers/google/cloud/operators/compute.py

```diff
@@ -68,6 +75,468 @@ def execute(self, context: 'Context'):
+        account from the list granting this role to the originating account (templated).
+    :param retry: A retry object used  to retry requests. If `None` is specified, requests
+        will not be retried.
+    :param timeout: The amount of time, in seconds, to wait for the request to complete.
+        Note that if `retry` is specified, the timeout applies to each individual attempt.
+    :param metadata: Additional metadata that is provided to the method.
+    """
+
+    operator_extra_links = (ComputeInstanceDetailsLink(),)
+
+    # [START gce_instance_insert_from_template_fields]
+    template_fields: Sequence[str] = (
+        'body',
+        'source_instance_template',
+        'project_id',
+        'zone',
+        'request_id',
+        'gcp_conn_id',
+        'api_version',
+        'impersonation_chain',
+    )
+    # [END gce_instance_insert_from_template_fields]
+
+    def __init__(
+        self,
+        *,
+        source_instance_template: str,
+        body: dict,
+        zone: str,
+        resource_id: Optional[str] = None,
+        project_id: Optional[str] = None,
+        request_id: Optional[str] = None,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = (),
+        gcp_conn_id: str = 'google_cloud_default',
+        api_version: str = 'v1',
+        validate_body: bool = True,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        self.source_instance_template = source_instance_template
+        self.body = body
+        self.zone = zone
+        self.resource_id = self.body["name"] if 'name' in body else resource_id
+        self.request_id = request_id
+        self._field_validator = None  # Optional[GcpBodyFieldValidator]
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+
+        if validate_body:
+            self._field_validator = GcpBodyFieldValidator(
+                GCE_INSTANCE_TEMPLATE_VALIDATION_PATCH_SPECIFICATION, api_version=api_version
+            )
+        self._field_sanitizer = GcpBodyFieldSanitizer(GCE_INSTANCE_FIELDS_TO_SANITIZE)
+        super().__init__(
+            resource_id=self.resource_id,
+            zone=zone,
+            project_id=project_id,
+            gcp_conn_id=gcp_conn_id,
+            api_version=api_version,
+            impersonation_chain=impersonation_chain,
+            **kwargs,
+        )
+
+    def _validate_all_body_fields(self) -> None:
+        if self._field_validator:
+            self._field_validator.validate(self.body)
+
+    def _validate_inputs(self) -> None:
+        if self.project_id == '':
+            raise AirflowException("The required parameter 'project_id' is missing")
+        if not self.zone:
+            raise AirflowException("The required parameter 'zone' is missing")
```

Comment:
I saw the exact same method in the other class in this file. Is there any way of inheriting from some common class if those classes share a lot of the code?

Reply:
Done
Was trying to use similar approach, but moved all repeated messages to parent class